### ISSUE TITLE
run error

### ISSUE THREAD

**Issue by likeucode** [facebook/fb-caffe-exts#4]
Hi, buddy
When i run 

> th torch2caffe/torch2caffe.lua --input /home/byx/openface2016/openface/models/openface/nn4.v1.t7 --prototxt nn4_small2.protptxt --caffemodel nn4_small2.caffemodel 10 3 96 96
> i meet a problem:
> ---
> 
> INFO:torch2caffe.lib_py:added layer caffe.SpatialConvolution_0: bottom=[data], top=[caffe.SpatialConvolution_0]
> /home/byx/torch/install/bin/luajit: ./torch2caffe/torch_layers.lua:40: Python error: opaque ref: call
> Traceback (most recent call last):
>   File "/home/byx/Downloads/fb-caffe-exts-master/torch2caffe/lib_py.py", line 73, in add_layer
>     bottom_edges = [int(bottom_edge) for bottom_edge in list(bottom_edges)]
> TypeError: 'NoneType' object is not iterable

stack traceback:
    [C]: in function 'add_layer'
    ./torch2caffe/torch_layers.lua:40: in function 'add'
    ./torch2caffe/torch_layers.lua:81: in function 'add'
    ./torch2caffe/lib.lua:161: in function 'convert'
    ./torch2caffe/lib.lua:168: in function 'main'
    torch2caffe/torch2caffe.lua:24: in main chunk
    [C]: in function 'dofile'
    .../byx/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
    [C]: at 0x00406670

---

and i don't know how to firgure it out, i need your help, please.

**Comment by ajtulloch** [comment]
Can you print out the `nn4.v1.t7` layer architecture you are trying to convert? Just print it from a th repl.

**Comment by likeucode** [comment]
Yes, this command can print nn4.v1.t7 layer architecture first, but then i got "add_layer" error

**Comment by happyharrycn** [comment]
I think this is because add_layer is not support by current version of torch2caffe. This layer is used by ResNet. Andrew, I wonder whether you plan to add it.

**Comment by ajtulloch** [comment]
At least the ResNet models we train can be converted - are you just using cudnn/nn.BatchNormalization and nn.CAddTable? If so, we already have this internally.

**Comment by austingg** [comment]
@ajtulloch seems no batchnormlization layer in torch_layers or caffe_layers.py 
Do you have any plan about adding this layer

**Comment by likeucode** [comment]
Hi, could you help me convert this torch [pre-trained model ](http://openface-models.storage.cmusatyalab.org/nn4.small2.v1.t7)to caffe model ? I tried many times , but got a failure at the end :(

**Comment by amrutha17** [comment]
@likeucode Have you been able to get through this? I am getting the same error when I try to convert a pre-trained 'SoundNet' model. Any help is appreciated.

**Comment by adhanaku** [comment]
@likeucode Have you been able to get through this? I am getting the same error when I try to convert a pre-trained 'SoundNet' model. Any help is appreciated.
